pkg/plugins/runtime/k8s/webhooks/injector/injector.go

```diff
@@ -189,14 +196,22 @@ func (i *KumaInjector) InjectKuma(ctx context.Context, pod *kube_core.Pod) error
 			pod.Annotations[metadata.KumaTrafficTransparentProxyConfig] = tproxyCfgYAML
 		}
 	} else { // this is legacy and deprecated - will be removed soon
-		if annotations, err = i.NewAnnotations(pod, meshName, logger); err != nil {
+		if annotations, err = i.NewAnnotations(pod, logger); err != nil {
 			return errors.Wrap(err, "could not generate annotations for pod")
 		}
 
 		for key, value := range annotations {
 			pod.Annotations[key] = value
 		}
 
+		if pod.Labels != nil {
+			pod.Labels[metadata.KumaMeshLabel] = meshName
+		} else {
+			pod.Labels = map[string]string{
+				metadata.KumaMeshLabel: meshName,
+			}
+		}
```

Comment:
also nit pick but I think

```suggestion
		if pod.Labels == nil {
			pod.Labels = map[string]string{}
		}
		pod.Labels[metadata.KumaMeshLabel] = meshName
```

is way less noisy